### ISSUE TITLE
All complications are blank

### ISSUE THREAD

**Issue by kellogg76** [itsMeDavidV/Watchla-Support#221]
Updated to latest version in the app store a few days ago, now all the complications for Watchla are blank on the watch.
If I edit the watch face, the description of the complications has the text but no icons.
Changing to another app's complication works, changing back to Range (for example) is blank again.
Refresh watch app on the phone does nothing.

**Comment by dbgerhard** [comment]
I have this issue as well. When I tap on the blank space on my watch where the complication should be, it opens the Watchla watch app and things seem to work ok.

**Comment by kellogg76** [comment]
> I have this issue as well. When I tap on the blank space on my watch where the complication should be, it opens the Watchla watch app and things seem to work ok.

Me too, but when I go back to the watch face by pressing the crown, the icons are all blank again. They seem to work when you go into the app but that doesn't get populated back to the complications.

**Comment by kellogg76** [comment]
Commenting to add that 3.5.2 released today doesn't fix anything.

**Comment by zunzster** [comment]
I found the issue is the background on all the Watchla Complications now show as mostly white (and hence barely viewable) when I have my Modular watchface with a rainbow background as opposed to black.

**Comment by itsMeDavidV** [comment]
This is a known issue and we are diligently working to address it. V3.5.2 addresses other issues raised. This will have its own update very shortly.

**Comment by AngryPotato99** [comment]
Watchla app compilations not showing on apple watch ultra 2

**Comment by claire078** [comment]
on 3.53 same to me. Battery status works, relevant vehicle status doesn’t.
Phone+watch apps are up to date.

**Comment by kellogg76** [comment]
It seems to have been fixed over the weekend, thanks.

**Comment by claire078** [comment]
> It seems to have been fixed over the weekend, thanks.

But not to me.
![6B65CB1B-4464-4A91-ADA0-576890DD3156_4_5005_c](https://github.com/itsMeDavidV/Watchla-Support/assets/46406097/5cd2b970-ba22-42f1-a498-6ae006246e71)
![983235ED-58F8-4E0F-90EB-AD93E543C2DC_4_5005_c](https://github.com/itsMeDavidV/Watchla-Support/assets/46406097/d9c7a620-35e6-4528-a8ba-d6bf2b08a8b7)

fyi Apple Watch and iPhone on 10.3, 17.3

**Comment by AngryPotato99** [comment]
Not fixed yet

**Comment by TeslaOwnerTips** [comment]
Latest app 3.5.5 (338)
I deleted and installed again 
Turned both watch and phone off/shutdown then on
Relevant vehicle status and sentry mode widgets blank. 

First image relevant vehicle status blank. 
Second image sentry mode status not showing 

![image](https://github.com/itsMeDavidV/Watchla-Support/assets/78335749/a8790e48-9478-465d-86b9-97c517fbeda7)
![image](https://github.com/itsMeDavidV/Watchla-Support/assets/78335749/9447ce95-8c1d-4d20-95ff-6e9d1c716eaa)

**Comment by TeslaOwnerTips** [comment]
Not showing, just a grey circle 
Flash lights
Honk horn
Lock/unlock
Open doors 
Open frunk
I stopped checking

**Comment by kellogg76** [comment]
Are these issues being actively worked on? The functionality has really taken a massive hit in recent "updates". If the data in the complications isn't being updated there's not really much point to the app. It doesn't report correct range, the open charge port and home link don't work either.

More often than not I get the spinny wheel icon filling the whole screen and then nothing.

**Comment by mbvisiontt** [comment]
It’s April 10th and still the same issue??

Having bought the app, pretty annoying to have it not work properly for months now.

**Comment by CassandraMarie** [comment]
Hi @zunzster, @dbgerhard, @kellogg76, @claire078, @mbvisiontt, @TeslaOwnerTips, @AngryPotato99, we apologize for the late reply. Are you still experiencing this issue? 

Complications should be visible with the latest version 3.8. You can check the version of both your iOS and watchOS by navigating to preferences and scrolling to the bottom of the screen.

**Comment by zunzster** [comment]
The complications are now visible when the background is not black for me so my issue is fixed.

**Comment by zunzster** [comment]
Hrmm... I'm still getting the issue that I needed to reauthenticate the watch app again (it seems to need that quite often - is that normal?) and after doing that, it seems connected to my tesla, but when I go into preferences to select vehicles to view my wife's tesla, no vehicles are listed so I can't switch.  That used to be quite useful prior to the rewrite to unlock her charging port.

The phone app can see both vehicles - but the watch app 'select vehicle' screen can't. No vehicles are listed, not even the one it's connected to. I selected Refresh watch app in the Phone app and that doesn't fix it either so I don't know how to fix this.

**Comment by mbvisiontt** [comment]
Hi,Yes they are visible, but NO, they are not updating (km indicator stuck on 240km, no matter what the real value is).Thanks for fixing this in the next update.BR, mbvisionttOp 2 aug 2024 om 02:19 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
Hi @zunzster, @dbgerhard, @kellogg76, @claire078, @mbvisiontt, @TeslaOwnerTips, @AngryPotato99, we apologize for the late reply. Are you still experiencing this issue?
Complications should be visible with the latest version 3.8. You can check the version of both your iOS and watchOS by navigating to preferences and scrolling to the bottom of the screen.

**Comment by CassandraMarie** [comment]
@zunzster, We believe you may be experiencing a known authentication issue. Could you please send your email and mention this GitHub issue via [https://watchla.app/contact/](https://watchla.app/contact/). Thank you

**Comment by CassandraMarie** [comment]
Hi @mbvisiontt, Could you please confirm that you are using version 3.8.1 of Watchla?

**Comment by mbvisiontt** [comment]
Confirmed. 3.8.1 since 2 months.Also did “refresh watch app” to no avail.Op 6 aug 2024 om 04:46 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt, Could you please confirm that you are using version 3.8.1 of Watchla?

**Comment by mbvisiontt** [comment]
And I just fully powered off and back on my phone and Watch, no solution.Also authenticated Watch app again, no solution.Op 6 aug 2024 om 06:03 heeft Martijn ***@***.***> het volgende geschreven:﻿Confirmed. 3.8.1 since 2 months.Also did “refresh watch app” to no avail.Op 6 aug 2024 om 04:46 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt, Could you please confirm that you are using version 3.8.1 of Watchla?

**Comment by mbvisiontt** [comment]
And… I just tried to switch the complication to range % and it was blank!! Nothing visible. Switched back to km complication and is visible but frozen on 240km as before.😕😕😕Op 6 aug 2024 om 06:11 heeft Martijn ***@***.***> het volgende geschreven:﻿And I just fully powered off and back on my phone and Watch, no solution.Also authenticated Watch app again, no solution.Op 6 aug 2024 om 06:03 heeft Martijn ***@***.***> het volgende geschreven:﻿Confirmed. 3.8.1 since 2 months.Also did “refresh watch app” to no avail.Op 6 aug 2024 om 04:46 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt, Could you please confirm that you are using version 3.8.1 of Watchla?

**Comment by CassandraMarie** [comment]
@mbvisiontt,  Can you send over your support code via [https://watchla.app/contact/](https://watchla.app/contact/) so we can investigate your case?

Support codes can be found by going to Preferences > Support/ Feedback > Support code
Please be sure to link to this issue when you reach out.

**Comment by mbvisiontt** [comment]
Hi,This is: [redacted]hanks.Op 6 aug 2024 om 19:54 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
@mbvisiontt,  Can you send over your support code via https://watchla.app/contact/ so we can investigate your case?
Support codes can be found by going to Preferences > Support/ Feedback > Support code
Please be sure to link to this issue when you reach out.

**Comment by pymiras** [comment]
Same here. All complications are blank. Very annoying. Are you still working on the app? This issue is open since January...

**Comment by mbvisiontt** [comment]
Hi,Not working still either… either blank or showing wrong value and not updating it.I shared my code from the app but heard nothing afterwards..! Please update and fix.Op 15 sep 2024 om 16:04 heeft pymiras ***@***.***> het volgende geschreven:﻿
Same here. All complications are blank. Very annoying. Are you still working on the app? This issue is open since January...

**Comment by CassandraMarie** [comment]
Hi @mbvisiontt, You seem to be using version 3.8.2,  please update both of your devices to version 3.8.3. Which contains background efficiency improvements that directly relate to your issue. 
Additionally, Watchla has registered your vehicle has been either slow to respond or slow to wake on different occasions. This is most likely due to the vehicle's internet connection and/or being in a state of sleep for an extended period of time.

**Comment by mbvisiontt** [comment]
Hi,It seems to be working at this point, I’m on 3.8.3. It has been working before for very short periods, so let’s hope it keeps working this time! ☺️Op 17 sep 2024 om 01:35 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt, You seem to be using version 3.8.2,  please update both of your devices to version 3.8.3. Which contains background efficiency improvements that directly relate to your issue.
Additionally, Watchla has registered your vehicle has been either slow to respond or slow to wake on different occasions. This is most likely due to the vehicle's internet connection and/or being in a state of sleep for an extended period of time.

**Comment by CassandraMarie** [comment]
@mbvisiontt, Good news! If you run into any more issues please reach out to us via [https://watchla.app/contact/](https://watchla.app/contact/).

**Comment by mbvisiontt** [comment]
Hi,Again the watch face isn’t updating, very disappointing for a paid app. Please check. Op 24 sep 2024 om 04:25 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
@mbvisiontt, Good news! If you run into any more issues please reach out to us via https://watchla.app/contact/.

**Comment by CassandraMarie** [comment]
Hi @mbvisiontt, Could you send over your support code via [https://watchla.app/contact/](https://watchla.app/contact/) so we can investigate your case?

**Comment by mbvisiontt** [comment]
Hi,It’s still the same issue as before… you got all the info from me in orevious mails under this case?!Op 8 nov 2024 om 18:49 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt, Could you send over your support code via https://watchla.app/contact/ so we can investigate your case?

**Comment by CassandraMarie** [comment]
Yes, I have your previous support code; however, support codes can sometimes change during updates. Since it’s been a few months, I wanted to confirm that we are looking into the correct one. Additionally are both of your devices up to date with the latest update, 3.9.2?

**Comment by mbvisiontt** [comment]
Hi,Yes both latest versions, problem persists. Logged out and in, changed complication and back, restarted both devices, resfreshed watch app. To no avail.Support code now says: MsKmgQuQx0Thx. Op 8 nov 2024 om 19:16 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
Yes, I have your previous support code; however, support codes can sometimes change during updates. Since it’s been a few months, I wanted to confirm that we are looking into the correct one. Additionally are both of your devices up to date with the latest update, 3.9.2?

**Comment by CassandraMarie** [comment]
Hi @mbvisiontt, we recently released an update that addresses complications. After reviewing your support code we noticed you are using an older version. Please update to 3.9.3 which addresses issues where complications may not be updated. Please reach back out if the issue persists. Thank you.

**Comment by mbvisiontt** [comment]
Hi,I’m on 3.9.3 both on phone and watch, problem persists.Op 11 nov 2024 om 21:10 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt, we recently released an update that addresses complications. After reviewing your support code we noticed you are using an older version. Please update to 3.9.3 which addresses issues where complications may not be updated. Please reach back out if the issue persists. Thank you.

**Comment by CassandraMarie** [comment]
When you open Watchla on your watch, does the vehicle data appear to be up to date?

**Comment by mbvisiontt** [comment]
Yes, that part seems to work well. Always has actually.Op 11 nov 2024 om 21:37 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
When you open Watchla on your watch, does the vehicle data appear to be up to date?

**Comment by CassandraMarie** [comment]
Hi @mbvisiontt, Could you let me know which complication isn’t updating and how accurate the displayed value is? For example, would you say the information is only slightly out of date or significantly inaccurate?
Could you please include the values?

**Comment by zunzster** [comment]
I have 3.9.3 and the watch app shows 98% and the complication still shows 94% - which would be from when the Tesla was charging it last night.

**Comment by mbvisiontt** [comment]
Hi,Values are completely off. Whether I change vehicles, doesn’t matter either; it has value 118 km currently, was from a car that once had that. If I switch to another that has 423, nothing happens. Still says 118.Only once every week or so (if I’m lucky), currently I’m not and 118 has been there longer, the value changes.Op 12 nov 2024 om 06:24 heeft Cassandra Marie ***@***.***> het volgende geschreven:﻿
What complication and how far off is the data? Would you say it’s only slightly off or significantly inaccurate?
Could you please include the values?

**Comment by itsMeDavidV** [comment]
Hi @mbvisiontt & @zunzster we made some changes to better address the issue of complications being slightly off (eg: shown range 195, actual range 200)

Please let us know if you see this behavior corrected over the next day or so or if you still see this unintended behavior persist.

**Comment by mbvisiontt** [comment]
Hi,Mine was way off, or better: not updated for days/weeks, and still is for now.Op 13 nov 2024 om 03:56 heeft David Villegas ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt & @zunzster we made some changes to better address the issue of complications being slightly off (eg: shown range 195, actual range 200)
Please let us know if you see this behavior corrected over the next day or so or if you still see this unintended behavior persist.

**Comment by mbvisiontt** [comment]
I just realised: I think my complication completely stopped working (to be clear: it was already bogus most of the time) when I added a 3rd car to my account.Please look into that if that explains something.Op 13 nov 2024 om 09:07 heeft Martijn ***@***.***> het volgende geschreven:﻿Hi,Mine was way off, or better: not updated for days/weeks, and still is for now.Op 13 nov 2024 om 03:56 heeft David Villegas ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt & @zunzster we made some changes to better address the issue of complications being slightly off (eg: shown range 195, actual range 200)
Please let us know if you see this behavior corrected over the next day or so or if you still see this unintended behavior persist.

**Comment by mbvisiontt** [comment]
I haven’t heard from you?I also experience the Watch complications resetting to the car with the lowest alphabet letter, so always to the D, whilst I select car names that begin with an M or W. Very annoying.Op 17 nov 2024 om 19:44 heeft Martijn ***@***.***> het volgende geschreven:﻿I just realised: I think my complication completely stopped working (to be clear: it was already bogus most of the time) when I added a 3rd car to my account.Please look into that if that explains something.Op 13 nov 2024 om 09:07 heeft Martijn ***@***.***> het volgende geschreven:﻿Hi,Mine was way off, or better: not updated for days/weeks, and still is for now.Op 13 nov 2024 om 03:56 heeft David Villegas ***@***.***> het volgende geschreven:﻿
Hi @mbvisiontt & @zunzster we made some changes to better address the issue of complications being slightly off (eg: shown range 195, actual range 200)
Please let us know if you see this behavior corrected over the next day or so or if you still see this unintended behavior persist.
